packages/developer-console-spa/src/pages/ProjectConsoleShell/index.tsx

```diff
@@ -1,18 +1,38 @@
 import { Flex, FlexItem } from '@patternfly/react-core';
-import { useContext } from 'react';
+import { useContext, useLayoutEffect } from 'react';
 import HeaderContextProvider from 'common/context/HeaderContext';
-import ProjectContextProvider, { ProjectContext } from 'common/context/ProjectContext';
+import ProjectContextProvider, {
+  ProjectContext,
+} from 'common/context/ProjectContext';
 import Header from 'common/components/Header';
 import Loader from 'common/components/Loader';
 import Sidebar from 'common/components/Sidebar';
+import { Redirect } from 'react-router-dom';
+import { isEmpty } from 'lodash';
 
 function ProjectConsoleShell(props: any) {
-  const { loading } = useContext(ProjectContext);
+  const { project, loading } = useContext(ProjectContext);
+
+  useLayoutEffect(() => {
+    if (!loading && isEmpty(project)) {
+      /* Artificially delaying the error notification for a cleaner effect */
+      setTimeout(() => {
```

Comment:
Delete this time out on unmount to avoid memory leak

Reply:
Removed that commit. The test for invalid project is not working for some reason. Will raise a separate PR for that fix later.